### DISCUSSION TITLE
ERROR: Aillio R1: unable to detach kernel driver

### DISCUSSION THREAD

**Discussion by Frothinger** [artisan-roaster-scope/artisan#1069] [General]
I am trying to connect my Aillio R1 roster to Artisan 2.8.2 on my Mac running Ventura 13.0.1

I keep getting the following error:

"Aillio R1: unable to detach kernel driver"

I have tried restarting my roaster and computer and tried several different startup procedures, but I still get this error.

Any suggestions or ideas?

I have attached a screenshot.
![Screenshot 2023-01-05 at 4 45 13 PM](https://user-images.githubusercontent.com/2790974/210907928-a4b3b5cd-02ba-4a88-9b8f-28f6bbc0b0b4.png)

**Comment by MAKOMO** [comment]
This seems to be due to a change in macOS 12 which is claiming all USB interfaces thus Artisan cannot attach. Sadly there is up to now no known way for a app (even if signed and notarized like Artisan) to have macOS to detach from that USB port.

You might be able to do that manually by issuing the following in a terminal before starting Artisan

```
sudo kextunload -b com.apple.driver.AppleUSBFTDI
```

We have no way to test this for a lack of access to that roaster. Could you try and report?

There is some more information [here](https://github.com/libusb/libusb/wiki/FAQ#How_can_I_run_libusb_applications_under_Mac_OS_X_if_there_is_already_a_kernel_extension_installed_for_the_device_and_claim_exclusive_access) in section _How can I run libusb applications under Mac OS X if there is already a kernel extension installed for the device and claim exclusive access?_

**Reply by Frothinger** [reply]
Thanks for the quick response.

Here is what I got when I used the command:

❯ sudo kextunload -b com.apple.driver.AppleUSBFTDI
Password:
Executing: /usr/bin/kmutil unload -b com.apple.driver.AppleUSBFTDI
Error Domain=KMErrorDomain Code=3 "Error occurred unloading extensions: Missing extension with identifier com.apple.driver.AppleUSBFTDI : Could not find: Extension with identifier 'com.apple.driver.AppleUSBFTDI' not found" UserInfo={NSLocalizedDescription=Error occurred unloading extensions: Missing extension with identifier com.apple.driver.AppleUSBFTDI : Could not find: Extension with identifier 'com.apple.driver.AppleUSBFTDI' not found}

**Reply by MAKOMO** [reply]
Thanks for testing. This is very funny. This indicates that the AppleUSBFTDI driver is not even loaded/available on macOS 13. Hm.

Anyhow, I have a good news for you. It seems the issue with the Aillio R1 got solved by just not trying to unload the USB driver. See Issue 1069.

The [current Continuous Builds](https://github.com/artisan-roaster-scope/artisan/releases/tag/continuous) contain the fix. As those builds are not signed you need to start the Artisan app on the first run via a right-click to its app icon and choosing Open from the menu.

**Reply by Frothinger** [reply]
2.8.3 seems to be working. Thank you!

Now that I am connected, let the roasting start!
